### PR TITLE
use __serialize instead of __sleep on CacheEntry

### PR DESCRIPTION
**Pull request by e-zannelli**
A proposed fix for https://github.com/Kevinrob/guzzle-cache-middleware/issues/162

### CODE REVIEW

**Comment by Kevinrob** [comment]
@e-zannelli Can you rebase it? We have now Gitub Actions with tests

**Comment by e-zannelli** [comment]
Rebased :bow:

**Comment by Kevinrob** [comment]
> Rebased 🙇

Thank you! The tests don't pass 😯

**Comment by e-zannelli** [comment]
I restored compatibility with php <7.4 by implementing \Serializable interface, tests pass now.

File: src/CacheEntry.php [hunk 1]
```diff
@@ -256,47 +257,72 @@ public function getAge()
...
-            ->withBody(
-                \GuzzleHttp\Psr7\Utils::streamFor((string) $this->request->getBody())
-            );
+        return serialize($this->__serialize());
     }
 
+    public function unserialize($data)
+    {
+        $this->__unserialize(unserialize($data));
+    }
```
Comment (Kevinrob) [inline src/CacheEntry.php:327]: What is the purpose of this function `public function unserialize($data)`? I don't find any call to it and it doesn't seem to be a "magic" PHP function.

Comment (jdreesen) [reply src/CacheEntry.php:327]: It's from PHPs legacy system for customized serialization, see: [The Serializable interface](https://www.php.net/manual/en/class.serializable.php)

**Comment by Kevinrob** [comment]
Thank @e-zannelli !

**Pull request merged**